src/makeInfinityAdder.js

```diff
@@ -7,6 +7,24 @@
 
 function makeInfinityAdder() {
   // write code here
-}
+  let sum = 0;
+  let currentArgs = [];
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum + currentArgs.reduce((acc, val) => acc + val, 0);
+
+      sum = 0;
+      currentArgs = [];
 
+      return result;
+    } else {
+      currentArgs = [...currentArgs, ...args];
+
+      return adder;
```

Comment:
```suggestion
      return result;
    }
      
    sum += number;

    return adder;
```
`else` is also redundant, as your code will not go further after return anyway